text/metrics/0108-naming-conventions.md

```diff
@@ -0,0 +1,23 @@
+# Metric instrument naming conventions
+
+## Purpose
+
+Names and labels for metric instruments are primarily how humans interact with metric data -- users rely on these names to build dashboards and perform analysis. The names and hierarchical structure need to be understandable and discoverable during routine exploration -- and this becomes critical during incidents.
+
+To ensure these goals and consistency in future metric naming standards, this outlines a meta-standard for these names.
+
+## Guidelines
+
+Metric names and labels exist within a single universe and a single hierarchy. Metric names and labels MUST be considered within the universe of all existing metric names. When defining new metric names and labels, consider the prior art of existing standard metrics and metrics from frameworks/libraries.
+
+Associated metrics SHOULD be nested together in a hierarchy based on their usage. Define a top-level hierarchy for common metric categories: for OS metrics, like CPU and network; for app runtimes, like GC internals. Libraries and frameworks should nest their metrics into a hierarchy as well. This aids in discovery and adhoc comparison. This allows a user to find similar metrics given a certain metric.
+
+The hierarchical structure of metrics defines the namespacing. Supporting OpenTelemetry artifacts define the metric structures and hierarchies for some categories of metrics, and these can assist decisions when creating future metrics.
+
+Common labels SHOULD be consistently named. This aids in discoverability and disambiguates similar labels to metric names.
+
```

Comment:
> Common labels SHOULD be consistently named

What does "common labels" mean? Common in what context/scope? A single service? An organization?

Reply:
I was hoping to avoid a debate over naming label keys. For example, you have a label key named "service" and have used it on some metrics, and I have a label key named "service" and have used it on some different metrics. How are we to know that those labels are not the same? The answer would be to add namespacing of labels. I recall the OpenCensus guidelines were to prefix your label names with a DNS prefix that you own. So I might have a `lightstep.com/service` label and you might have an `uber.com/service` label. For this to create a good user experience, I'd like the DNS prefix to not display by default. What would you like to see, @yurishkuro?

Reply:
The way I read this guidance is: if we have a label that should be added to many different categories of metric instrument, and that label's semantic meaning is the same across all those categories, its name should be consistent.  

The most obvious example I can think of would be _status_, whose value will be a CanonicalSpanStatus.  

As a user, I would find it intuitive when searching my metrics in my UI to always find the success/failure information under the same `status` label.

I'm not sure I understand the example `service` label.  Would it be the name of the service being instrumented?  If so, perhaps we would want some semantic conventions around how to apply Resource attributes as metric labels.

If this is the case, I'm not sure if we need to change this line. Is this guidance not clear enough? What wording would make our meaning more clear?

Reply:
I think it's safe to say we can merge this and debate this topic again as we modify the specification.